framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -592,195 +497,285 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
+     *
+     * <p>If this method returns false, the {@code expression} is not parsed and will appear in the
+     * dependent types annotation as is.
      *
-     * @param expression a Java expression
-     * @param context the context
-     * @param localVarPath if non-null, the expression is parsed as if it were written at this
-     *     location
-     * @return the standardized version of the Java expression
+     * @param expression an expression string in a dependent types annotation
+     * @return whether or not {@code expression} should be parsed
      */
-    protected String standardizeString(
-            String expression, JavaExpressionContext context, @Nullable TreePath localVarPath) {
-        if (DependentTypesError.isExpressionError(expression)) {
-            return expression;
-        }
-        JavaExpression result;
-        try {
-            result = JavaExpressionParseUtil.parse(expression, context, localVarPath);
-        } catch (JavaExpressionParseUtil.JavaExpressionParseException e) {
-            return new DependentTypesError(expression, e).toString();
-        }
-        if (result == null) {
-            return new DependentTypesError(expression, /*error message=*/ " ").toString();
-        }
-
-        return result.toString();
+    protected boolean shouldParseExpression(String expression) {
+        return !DependentTypesError.isExpressionError(expression);
     }
 
     /**
-     * Viewpoint-adapts Java expressions in an annotation. If the annotation is not a dependent type
-     * annotation, returns null.
+     * Create a new annotation of the same type as {@code originalAnno} using the provided {@code
+     * elementMap}.
      *
-     * @param context information about any receiver and arguments
-     * @param localVarPath if non-null, the expression is parsed as if it were written at this
-     *     location
-     * @param anno the annotation to viewpoint-adapt
-     * @param removeErroneousExpressions if true, remove erroneous expressions rather than
-     *     converting them into an explanation of why they are illegal
-     * @return the viewpoint-adapted annotation, or null if no viewpoint-adaption is needed
+     * @param originalAnno the annotation passed to {@link
+     *     #convertAnnotationMirror(StringToJavaExpression, AnnotationMirror)}
+     * @param elementMap a mapping of element names of {@code originalAnno} to {@code
+     *     JavaExpression}s
+     * @return an annotation created from {@code elementMap}
      */
-    public AnnotationMirror standardizeAnnotationIfDependentType(
-            JavaExpressionContext context,
-            @Nullable TreePath localVarPath,
-            AnnotationMirror anno,
-            boolean removeErroneousExpressions) {
-        if (!isExpressionAnno(anno)) {
-            return null;
-        }
-
+    protected AnnotationMirror buildAnnotation(
+            AnnotationMirror originalAnno, Map<String, List<JavaExpression>> elementMap) {
```

Comment:
This implementation looks like it works if all of an annotation's elements are Java expressions.  But what if some elements are Java expressions and others are not?  The others seem to be dropped by this implementation.